remotes/docker/resolver.go

```diff
@@ -364,7 +364,8 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 				size = bodyReader.bytesRead
 			}
 			// Prevent resolving to excessively large manifests
-			if size > MaxManifestSize {
+			isManifest := u[0] == "manifests"
```

Comment:
It looks a bit hacky and initially I had something like `images.IsManifestType(contentType) || images.IsIndexType(contentType)`.
Then I looked at the [distribution spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#definitions) and it actually says that: "Manifest: a JSON document uploaded via the manifests endpoint. (...)", so I went for this one instead.